pkg/adapter/awscodecommitsource/adapter.go

```diff
@@ -140,32 +140,44 @@ func (a *adapter) Start(stopCh <-chan struct{}) error {
 		a.logger.Errorw("Failed to process pull requests", "error", err)
 	}
 
-	//range time.Tick(time.Duration(syncTime) * time.Second)
-	for {
-		if strings.Contains(a.gitEvents, pushEventType) {
-			err := a.processCommits()
-			if err != nil {
-				a.logger.Errorw("Failed to process commits", "error", err)
-			}
-		}
+	backoff := common.NewBackoff()
 
-		if strings.Contains(a.gitEvents, prEventType) {
-			pullRequests, err := a.preparePullRequests()
-			if err != nil {
-				a.logger.Errorw("Failed to process pull requests", "error", err)
-			}
+	err = backoff.Run(stopCh, func() func(context.Context) (bool, error) {
```

Comment:
I'm having a hard time wrapping my head around that `func() func(context.Context) (bool, error) { /* */ }()`. Why don't we pass a `func(context.Context) (bool, error)` directly?

Reply:
We do now, thanks!